### ISSUE TITLE
Update broken link build failure threshold logic

### ISSUE THREAD

**Issue by acrollet** [department-of-veterans-affairs/va.gov-cms#5050]
## User Story

As a CMS content editor, I do not wish a single broken link on unrelated content to fail the content build, in order that I may publish important updates in a timely fashion.

As a CMS team engineer, I should be notified in the #vfs-platform-builds slack channel about broken links so that I know I am responsible for triage.

## Implementation

Option 2 is to be preferred if possible, see [this slack thread](https://dsva.slack.com/archives/CT4GZBM8F/p1619019011219600) for more background.

### Option 1

The content build jenkins job should be modified to parse the broken links CSV and only fail the build if:

- the page on which the broken link occurs is the homepage, OR
- there are more than 10 broken links in total

Slack notifications should continue to be sent for broken links, and should specify whether or not the build was failed.

### Option 2

Modify the link checker in https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/modify-dom/check-broken-links/ to read a config file that specifies the paths that should be ignored, those paths that cause a build failure, and the threshold number of non-critical links that should cause a build failure. Ideally this work would also create more easily human-readable output that can be included in slack notifications. The format could potentially look something like this:
```yaml
ignored_routes:
  - '*.asp'
error_paths:
  - '/index.html'
  - '/other_really_important_path/index.html'
# Mark links as errored when more than this total are found:
error_threshold: 10
```
The jenkins job would then need to be modified to understand the new output and fail the build only when appropriate.

## Acceptance Criteria

- [x] any broken link on the homepage continues to fail the build
- [x] fewer than 11 broken links on other pages does not fail the build
- [x] Broken link slack notifications specify whether or not the build was failed
- [x] Broken link notifications are delivered to #vfs-platform-builds slack channel
- [ ] Broken link triage process is updated to reflect new process: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/cms/helpdesk/broken-link-process.md

**Comment by mchelen-gov** [comment]
preferably this logic would exist JS build step rather than Jenkins, for sake of consistency and forwards compatibility with new CI/CD. not sure if that is prohibitively difficult though?

**Comment by mchelen-gov** [comment]
@acrollet that seems like a reasonable approach, the other place that has a bunch of broken link logic is https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/modify-dom/check-broken-links/index.js so i was wondering if it's possible to include it there?

**Comment by mchelen-gov** [comment]
option 2 might be worth trying, if it's too difficult then go with option 1 and spin off followup ticket

**Comment by acrollet** [comment]
PR to update the slack channel: https://github.com/department-of-veterans-affairs/vets-website/pull/17020

**Comment by mchelen-gov** [comment]
@ncksllvn sounds like this is believed to be finished, and we're waiting on a broken link to confirm?

can you link any associated fixes to this ticket

**Comment by ncksllvn** [comment]
Original PR for these changes, https://github.com/department-of-veterans-affairs/vets-website/pull/16972

Patch after discovering 1 broken link was still failing releases, https://github.com/department-of-veterans-affairs/vets-website/pull/17010

Patch after discovering the preview server was affected, https://github.com/department-of-veterans-affairs/vets-website/pull/17017

**Comment by mchelen-gov** [comment]
sounds like https://github.com/department-of-veterans-affairs/va.gov-cms/issues/5204 is also a followup to this

**Comment by ncksllvn** [comment]
Ugh, yeah, it is

**Comment by ncksllvn** [comment]
another PR...https://github.com/department-of-veterans-affairs/vets-website/pull/17046

**Comment by ncksllvn** [comment]
And one more, https://github.com/department-of-veterans-affairs/vets-website/pull/17044/files#diff-2cc6a42e83abdf9666365745cd63a9a93da477f58972350809e0b2acd32dfaceR199 

^ this one is the second try at getting Jenkins to _not_ throw a serialization error after sending the Slack message when a broken link is spotted. I think I got it this time.

**Comment by ncksllvn** [comment]
this has been validated

5 broken links flagged, but the content release was allowed out, https://dsva.slack.com/archives/C0MQ281DJ/p1620155124407900

**Comment by acrollet** [comment]
Let's wait for the first instance of ten+ broken links and then we can close this out. (Likely to happen tomorrow with continuing VAMC rollouts, due to race condition issues during bulk publishing)
